### PR TITLE
refactor: add plugin subcommand

### PR DESCRIPTION
**Pull request by afdesk**
## Checklist
- [x] I've read the [guidelines for contributing](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/) to this repository.
- [x] I've followed the [conventions](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/#title) in the PR title.
- [ ] I've added tests that prove my fix is effective or that my feature works.
- [ ] I've updated the [documentation](https://github.com/aquasecurity/trivy/blob/main/docs) with the relevant information (if needed).
- [ ] I've added usage information (if the PR introduces new options)
- [ ] I've included a "before" and "after" example to the description (if the PR is a user interface change).

### CODE REVIEW

File: pkg/commands/app.go [hunk 1]
```diff
@@ -70,11 +69,47 @@ func NewApp(version string) *cobra.Command {
 		NewSBOMCommand(globalFlags),
 		NewVersionCommand(globalFlags),
 	)
-	rootCmd.AddCommand(plugin.LoadCommands()...)
+	rootCmd.AddCommand(loadPluginCommands()...)
 
 	return rootCmd
 }
 
+func runPluginWithArgs(ctx context.Context, url string, args []string) error {
```
Comment (knqyf263) [inline pkg/commands/app.go:77]: We can move it to the `plugin` package.

Comment (afdesk) [reply pkg/commands/app.go:77]: sure, good idea! thanks

File: pkg/commands/app.go [hunk 2]
```diff
@@ -381,14 +403,94 @@ func NewPluginCommand() *cobra.Command {
...
+			Aliases:               []string{"i"},
+			Short:                 "install a plugin",
+			SilenceErrors:         true,
+			DisableFlagsInUseLine: true,
+			Args:                  cobra.ExactArgs(1),
+			RunE: func(cmd *cobra.Command, args []string) error {
+				if _, err := plugin.Install(cmd.Context(), args[0], true); err != nil {
+					return xerrors.Errorf("plugin install error: %w", err)
+				}
+				return nil
```
[the remark this answers is not in the document]
Comment (knqyf263) [reply pkg/commands/app.go:417]: The plugin package should not know the implementation of CLI, urfave/cli or spf13/cobra. I don't want to pass cobra.Command to "plugin"

[the remark this answers is not in the document]
Comment (afdesk) [reply pkg/commands/app.go:417]: `plugin.Install()` gets `context.Context`

**Pull request merged**